packages/rollup-core/test/app/verifier.spec.ts

```diff
@@ -0,0 +1,376 @@
+/* External Imports */
+import { TestUtils } from '@eth-optimism/core-utils'
+
+/* Internal Imports */
+import { VerifierDataService } from '../../src/types/data'
+import { FraudProver, VerificationCandidate } from '../../src/types'
+import { Verifier } from '../../src/app'
+
+class MockVerifierDataService implements VerifierDataService {
+  public readonly verificationCandidates: VerificationCandidate[] = []
+  public readonly batchesVerified: number[] = []
+
+  public async getVerificationCandidate(): Promise<VerificationCandidate> {
+    return this.verificationCandidates.shift()
+  }
+
+  public async verifyBatch(batchNumber): Promise<void> {
+    this.batchesVerified.push(batchNumber)
+  }
+}
+
+interface Fraud {
+  batchNumber: number
+  batchIndex: number
+}
+class MockFraudProver implements FraudProver {
+  public readonly provenFraud: Fraud[] = []
+
+  public async proveFraud(
```

Comment:
Nice and we'll circle back with @kfichter for the final param requirements when the time comes